### PR TITLE
Expand operator instance screen information into OLM UI

### PR DESCRIPTION
**Pull request by dmvolod**
This fix will be able to control fields dependencies and standard and custom CR selection for the OLM UI, i.e.
image

image

### CODE REVIEW

**Comment by oraNod** [comment]
@dmvolod Wait. Does this mean we can control field labels and descriptions in OLM UI now?

**Review by ryanemerson — APPROVED** [review]

File: deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -80,33 +80,145 @@ spec:
...
+            path: service.sites.locations[].secretName
+            x-descriptors:
+              - 'urn:alm:descriptor:io.kubernetes:Secret'
+          - displayName: Node port
+            description: Node port to which the load balancer service forwards traffic
+            path: service.sites.local.expose.nodePort
+            x-descriptors:
+              - 'urn:alm:descriptor:com.tectonic.ui:number'
+              - 'urn:alm:descriptor:com.tectonic.ui:fieldDependency:service.sites.local.expose.type:NodePort'
+          - displayName: Cache replication factor
```
Comment (oraNod) [inline deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml:184]:
```suggestion
          - displayName: Number of owners
```

File: deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -80,33 +80,145 @@ spec:
...
+            description: Configures replicas for each cache entry
```
Comment (oraNod) [inline deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml:185]:
```suggestion
            description: Controls cache replication factor, or number of copies for each entry.
```

File: deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml [hunk 3]
```diff
@@ -80,33 +80,145 @@ spec:
...
+            path: service.replicationFactor
+            x-descriptors:
+              - 'urn:alm:descriptor:com.tectonic.ui:number'
+          - description: Enable/disable container ephemeral storage
+            displayName: Container ephemeral storage
+            path: service.container.ephemeralStorage
+            x-descriptors:
+              - 'urn:alm:descriptor:com.tectonic.ui:booleanSwitch'
+          - displayName: Storage size
+            description: The size of the storage to use in the persistent volume claim
```
Comment (oraNod) [inline deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml:195]:
```suggestion
            description: Sets the amount of storage for the persistent volume claim.
```

File: deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml [hunk 4]
```diff
@@ -80,33 +80,145 @@ spec:
...
+            path: service.container.storage
+            x-descriptors:
+              - 'urn:alm:descriptor:text'
+          - displayName: Storage class name
+            description: The name of the storage class object to use for the persistent volume claim
```
Comment (oraNod) [inline deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml:200]: Names the storage class object for persistent volume claims.

File: deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml [hunk 5]
```diff
@@ -80,33 +80,145 @@ spec:
...
+            path: service.container.storageClassName
+            x-descriptors:
+              - 'urn:alm:descriptor:io.kubernetes:StorageClass'
+              - 'urn:alm:descriptor:com.tectonic.ui:fieldDependency:service.container.ephemeralStorage:false'
+          - displayName: Expose host name for the Route
```
Comment (oraNod) [inline deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml:205]:
```suggestion
          - displayName: Route hostname
```

File: deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml [hunk 6]
```diff
@@ -80,33 +80,145 @@ spec:
...
+            description: Specifies the hostname where Infinispan is exposed
```
Comment (oraNod) [inline deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml:206]:
```suggestion
            description: Defines the network hostname for your Infinispan cluster.
```

File: deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml [hunk 7]
```diff
@@ -80,33 +80,145 @@ spec:
...
+            path: expose.host
+            x-descriptors:
+              - 'urn:alm:descriptor:com.tectonic.ui:text'
+              - 'urn:alm:descriptor:com.tectonic.ui:fieldDependency:expose.type:Route'
+          - displayName: Persistent volume claim name
```
Comment (oraNod) [inline deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml:211]:
```suggestion
          - displayName: Persistent volume claim (PVC) name
```

File: deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml [hunk 8]
```diff
@@ -80,33 +80,145 @@ spec:
...
+            description: Name of the persistent volume claim with custom libraries
```
Comment (oraNod) [inline deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml:212]:
```suggestion
            description: Names the PVC that holds custom libraries.
```

File: deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml [hunk 9]
```diff
@@ -80,33 +80,145 @@ spec:
...
+            path: dependencies.volumeClaimName
+            x-descriptors:
+              - 'urn:alm:descriptor:io.kubernetes:PersistentVolumeClaim'
         statusDescriptors:
           - description: The current pods
```
Comment (oraNod) [inline deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml:217]:
```suggestion
          - description: Displays the current status of pods.
```

File: deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml [hunk 10]
```diff
@@ -80,33 +80,145 @@ spec:
...
             displayName: Pods Status
```
Comment (oraNod) [inline deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml:218]:
```suggestion
            displayName: Pod Status
```

File: deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml [hunk 11]
```diff
@@ -80,33 +80,145 @@ spec:
...
             x-descriptors:
-              - "urn:alm:descriptor:com.tectonic.ui:podStatuses"
-      - description: Restore is the Schema for the restores API
-        kind: Restore
-        name: restores.infinispan.org
-        version: v2alpha1
-        displayName: Restore
+              - 'urn:alm:descriptor:com.tectonic.ui:podStatuses'
   description: |
       Infinispan is a distributed, in-memory data store that increases application performance and delivers open-source capabilities to handle demanding use cases.
```
Comment (oraNod) [inline deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml:223]:
```suggestion
      Infinispan is a distributed in-memory data store built on open-source technology.
```

**Review by oraNod — CHANGES_REQUESTED** [review]
@dmvolod Would it be possible to see more before/after screengrabs with this PR please? 

This is great work by the way! It should really help improve UX experience and improve quality.

File: deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml [hunk 12]
```diff
@@ -80,33 +80,145 @@ spec:
...
+            x-descriptors:
+              - 'urn:alm:descriptor:io.kubernetes:StorageClass'
+      - description: Restore is the Schema for the restores API
+        kind: Restore
+        name: restores.infinispan.org
+        version: v2alpha1
+        displayName: Restore
+        specDescriptors:
+          - displayName: Cluster name
+            description: Infinispan cluster name
```
[the remark this answers is not in the document]
Comment (dmvolod) [reply deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml:101]: @oraNod `description` and `displayName` are placed around the field but in separate places (below and above the field enter place) Please see at the screen. displayName is a name of the fields by default and can generated from it and description is additional info we can say customer.
image

[the remark this answers is not in the document]
Comment (oraNod) [reply deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml:101]: Thanks @dmvolod that helps! I'll have another pass at the strings tomorrow and will update some of the comments.

File: deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml (hunk 10 above)
Comment (dmvolod) [reply deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml:218]: This is not single Pod status, it's pie-chart diagram for all of the single Infinispan CR Pods

File: deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml (hunk 11 above)
Comment (dmvolod) [reply deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml:223]: Could you please review whole description, but I'm not owner and author if it

File: deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml [hunk 13]
```diff
@@ -80,35 +80,147 @@ spec:
...
+            path: service.container.ephemeralStorage
+            x-descriptors:
+              - 'urn:alm:descriptor:com.tectonic.ui:booleanSwitch'
+          - displayName: Storage size
+            description: Sets the amount of storage for the persistent volume claim.
+            path: service.container.storage
+            x-descriptors:
+              - 'urn:alm:descriptor:text'
+          - displayName: Storage class name
+            description: The name of the storage class object to use for the persistent volume claim
```
Comment (dmvolod) [inline deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml:200]:
```suggestion
            description: Names the storage class object for persistent volume claims.
```

**Comment by dmvolod** [comment]
Thanks @oraNod for review, I will provide OCP instance with changes to you soon internally

File: deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml (hunk 11 above)
Comment (oraNod) [reply deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml]: Thanks @dmvolod I remember writing that back in the early days of the operator. I kind of feel like we can trim it down but maybe it should stay the same to match what's in the CSV for the bundle: http://pkgs.devel.redhat.com/cgit/containers/datagrid-8-prod-operator-bundle/tree/manifests/datagrid.clusterserviceversion.yaml?h=datagrid-8.1-rhel-8#n120 

@rigazilla and @ryanemerson do either of you have an opinion on this description?

Comment (rigazilla) [reply deploy/olm-catalog/infinispan-operator.clusterserviceversion.yaml]: @oraNod, feel free to improve the description, imo staying close to RHDG 8.1 is not a requirement for master branch.

**Comment by ryanemerson** [comment]
@oraNod @dmvolod What's the status of this PR?

**Review by oraNod — APPROVED** [review]
LGTM

**Comment by oraNod** [comment]
@ryanemerson Thanks for the bump on this PR. I think it should be fine to merge. 

@dmvolod made this comment in a separate email: "will be temporary solution, when we will use a newer Operator SDK (1.0 and later) will be possible to generate these descriptions and meta inside Golang code
But will be possible to move out work to the code more easily than create it from scratch as descriptions and comments in the code are not ideal."

I think we should just go with what we have here as it is an improvement, in my opinion, on the generated field labels and descriptions. When we get to the new Operator SDK and can properly externalize microcopy strings, I'd like to revisit and put more effort into inline help for the OLM UI.

**Review by rigazilla — APPROVED** [review]

**Comment by rigazilla** [comment]
merged thanks, @dmvolod !

**Pull request merged**